src/cstd.h

```diff
@@ -304,6 +304,46 @@ stdnone_of(void *arr, size_t stride, size_t len, int (boolfunc)(const void *))
   return 1;
 }
 
+// Swaps element a with element b
+// complains if either is null for some reason
+#define __STDSWAP(type)                     \
+        void stdswap_##type(type* _val1, type* _val2) \
+        {                                             \
+          if (_val1 == NULL || _val2 == NULL)         \
+            __STD_PANIC("Error: null pointer detected in stdswap_%s", #type); \
+          type tmp = *_val1;                          \
+          *_val1 = *_val2;                            \
+          *_val2 = tmp;                               \
+        }
+
+
+typedef int (*CompareFunction)(const void *, const void *);
+
+const void 
+*is_sorted_until(const void *first, const void *last, CompareFunction compare) 
+{
```

Comment:
Make sure to use 2 spaces as indents instead of 4. I know im a bad person lol. Also there's a lot of whitespace at the end of some lines.